cetmix_tower_server/models/cx_tower_template_mixin.py

```diff
@@ -45,6 +45,32 @@ def get_variables_from_code(self, code):
         undeclared_variables = meta.find_undeclared_variables(ast)
         return list(undeclared_variables) if undeclared_variables else []
 
+    def _compute_variable_ids_common(self, records, field_names):
+        """
+        Compute and assign variable_ids for a set of records based on
+        references from the specified fields.
+
+        Args:
+            records (recordset): The recordset for which variable_ids will be computed.
+            field_names (list): Field names to extract variable references.
+
+        """
+        variable_obj = self.env["cx.tower.variable"]
+        for record in records:
+            all_references = set()
+            for field_name in field_names:
+                value = getattr(record, field_name, None)
+                if value:
+                    all_references.update(self.get_variables_from_code(value))
+
+            if all_references:
+                variables = variable_obj.search(
+                    [("reference", "in", list(all_references))]
+                )
+                record.variable_ids = [(6, 0, variables.ids)]
+            else:
+                record.variable_ids = [(5, 0, 0)]
+
```

Comment:
@GabbasovDinar Since only `cx.tower.file` and `cx.tower.command` inherit from `cx.tower.template.mixin`, executing `record._prepare_variable_commands(["code", "path"])` will not be possible for all models. Do you have any alternative suggestions to address this?

Reply:
@angelinaanaki ok,

maybe in this case we can modify the method:

```python

def _prepare_variable_commands(self, field_names, force_record=None):
        """
        TODO: Add docstring why and when we use the force_record param
        """
        record = force_record or self
        record.ensure_one()

        all_references = set()
        for field_name in field_names:
            value = getattr(record, field_name, None)
            if value:
                all_references.update(self.get_variables_from_code(value))

        if all_references:
            variables = self.env["cx.tower.variable"].search(
                [("reference", "in", list(all_references))]
            )
            command = [(6, 0, variables.ids)]
        else:
            command = [(5, 0, 0)]

        return command

```

when not inherited the mixin:

```python
    @api.depends("code")
    def _compute_variable_ids(self):
        template_mixin_obj = self.env["cx.tower.template.mixin"]
        for record in self:
            record.variable_ids = template_mixin_obj._prepare_variable_commands(["code", "path"], force_record=record)
```

when inherited:

```python
record.variable_ids = record._prepare_variable_commands(["code", "path"])
```